nomium/shares-logger/src/storage/clickhouse/service.rs

```diff
@@ -171,12 +172,82 @@ impl ShareStorage<BlockFound> for ClickhouseBlockStorage {
     async fn init(&self) -> Result<(), ClickhouseError> {
         self.ensure_blocks_table_exists().await
     }
-
     async fn store_share(&mut self, block: BlockFound) -> Result<(), ClickhouseError> {
-        self.batch.push(block);
+        let block_hash_hex = hex::encode(&block.block_hash);
```

Comment:
Adding logic for block verification with retries using Bitcoin RPC within store_share function.